libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1292,15 +1301,34 @@ string ABIFunctions::abiDecodingFunctionByteArray(ArrayType const& _type, bool _
 	});
 }
 
+string ABIFunctions::abiDecodingFunctionCalldataStruct(StructType const& _type)
+{
+	string functionName =
+		"abi_decode_" +
+		_type.identifier();
```

Comment:
Better take care that this can never cause a collision with `abiDecodingFunctionStruct`

Reply:
I'll assert that the type is in calldata here and not in calldata in ``abiDecodingFunctionStruct`` - that'll make sure that the type identifier is in fact different.